contrib/mac/framework/julia-config.jl

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env julia
+# This file is a part of Julia. License is MIT: http://julialang.org/license
+
+const options =
+[
+    "--cflags",
+    "--ldflags",
+    "--ldlibs"
+];
```

Comment:
It is possible to do symlinks to a traditional unix hierarchy, but the Apple code signing documentation hints that this may break in the future.  In my experience with code signing over the last several years, each new OS version has become more strict on the code signing rules.  Beyond code signing concerns, sandboxing (whether the App Sandbox or not) is strict about the layout of code and resources and code identifiers.

Other cross platform projects that choose to offer a framework target mostly fall back on a symlink to unix hier approach, but I believe that Apple's documentation and my experience with code signing and sandbox strongly suggest following Apple's recommendations for framework layout.

If `julia-config.jl` knows how to handle frameworks, then general use code can rely on its output to compile and link against a Julia framework.  Cmake knows how to transparently handle frameworks as well.